### PR TITLE
feat!: Add support for data tiering attribute, bump minimum supported versions of Terraform and AWS provider

### PR DESCRIPTION
**Pull request by typat**
## Description
Without the ability to set the data tiering parameter during cluster creation, data tiering node types are not able to be used. The data tiering attribute is supported by the [AWS provider](https://github.com/hashicorp/terraform-provider-aws/pull/28022) since v4.42.0

VPC version was updated to 4 to get validate passing due to enable_classiclink being removed in AWS Provider 5.0.0.

## Motivation and Context
This change allows the use of data tiering node groups and supports https://github.com/hashicorp/terraform-provider-aws/pull/28022.

## Breaking Changes
Yes, lots of breaking changes due to major version increase in AWS provider. 

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 3.71"
+      version = ">= 4.42.0"
```
Comment (bryantbiggs) [inline versions.tf:7]: Since this is now a breaking change, lets just go all the way up to v5.0 of the AWS provider

Comment (typat) [reply versions.tf:7]: sounds good!

File: examples/data-tiering/main.tf [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  # Cluster
+  name        = local.name
+  description = "Example MemoryDB cluster"
+
+  engine_version             = "7.0"
+  auto_minor_version_upgrade = true
+  node_type                  = "db.r6gd.xlarge"
+  num_shards                 = 2
+  num_replicas_per_shard     = 2
+  data_tiering               = true
```
Comment (bryantbiggs) [inline examples/data-tiering/main.tf:38]: we can add this to the existing example - no need for a whole new example just for this

Comment (typat) [reply examples/data-tiering/main.tf:38]: perfect. i debated this but wasnt sure

File: versions.tf (hunk 1 above)
Comment (typat) [reply versions.tf]: ah missed your second comment

**Review by bryantbiggs — APPROVED** [review]
Thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 2.0.0 :tada:

**Comment by typat** [comment]
thanks for the quick feedback!

**Pull request merged**